File: library/src/test/java/me/proxer/library/enums/LanguageTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class LanguageTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final Language language = api.moshi().adapter(Language.class).fromJson("\"de\"");

        assertThat(language).isSameAs(Language.GERMAN);
    }

    @Test
    void testFallback() throws IOException {
        final Language language = api.moshi().adapter(Language.class).fromJson("\"XYZ\"");

        assertThat(language).isSameAs(Language.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/LanguageTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class LanguageTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(Language::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"de\"") shouldBe Language.GERMAN
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"XYZ\"") shouldBe Language.OTHER
    }
}
